server/datastore/mysql/app_configs.go

```diff
@@ -23,6 +29,32 @@ func (d *Datastore) AppConfig() (*kolide.AppConfig, error) {
 	return info, nil
 }
 
+func (d *Datastore) ManageHostExpiryEvent(hostExpiryEnabled bool, hostExpiryWindow int) error {
+	rows, err := d.db.Query("SELECT @@event_scheduler")
+	if err != nil {
+		return err
+	}
+	if !rows.Next() {
+		return errors.New("Error detecting event scheduler status.")
+	}
+	var value string
+	if err := rows.Scan(&value); err != nil {
+		return err
+	}
+	if value != "ON" {
+		return errors.New("Event Scheduler must be enabled for configuring Host Expiry.")
+	}
+
+	if !hostExpiryEnabled {
+		_, err := d.db.Exec(fmt.Sprintf("DROP EVENT IF EXISTS %s", HostExpiryEvent))
+		return err
+	}
+
+	_, err = d.db.Exec(
+		fmt.Sprintf("CREATE EVENT IF NOT EXISTS %s ON SCHEDULE EVERY 1 HOUR ON COMPLETION PRESERVE DO DELETE FROM HOSTS WHERE DATE_ADD(day, -%d, NOW()) < updated_at", HostExpiryEvent, hostExpiryWindow))
```

Comment:
Please use parameters here as well.

Reply:
A couple things:

1. Prefer to use `seen_time` for the last time the host checked in.

2. Does this `DATE_ADD` syntax work for you? I get
```
Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '-2, NOW()) < updated_at' at line 1
```
It looks to me like the correct syntax (based on https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-add) might be `WHERE seen_time < DATE_SUB(NOW(), INTERVAL 2 DAYS)` (reversing the comparison and using `DATE_SUB` makes it easier for me to read).

Reply:
Ah, cool, thank you! My SQL is shoddy at best :+1:

Reply:
Turns out that `CREATE EVENT` is not a supported prepared statement. I've just added the event name in the statements (i.e. `DROP EVENT IF EXISTS host_expiry`), but I can't seem to get around the `INTERVAL ? DAY` bit - thoughts on using `fmt.Sprintf` just for the string substitution of the integer window value?

Reply:
https://dev.mysql.com/doc/refman/8.0/en/sql-syntax-prepared-statements.html - See "SQL Syntax Allowed in Prepared Statements"

Reply:
Well that is appalling that prepared statements don't support creating events. We do have a couple mitigations here:

1) The user-controlled input is formatted as an integer in the `fmt.Sprintf`.
2) Only admins may modify the configuration value.